### PR TITLE
Automate publishing docs on CirclrCI

### PR DESCRIPTION
**Pull request by v-jizhang**
Automate publishing docs on CirclrCI

### CODE REVIEW

**Comment by zpao** [comment]
Let's do this on a Linux machine.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1226,6 +1226,36 @@ jobs:
             python3 ./tools/release/publish_release.py --no-build-deb --no-build-homebrew --github-token ${GITHUB_TOKEN} --chocolatey-token ${CHOCO_TOKEN} --use-existing-release --version ${VERSION} --output-dir artifacts
           shell: bash.exe
 
+  publish_docs:
+    environment:
+      <<: *linux_environment
+    working_directory: "/home/circleci/buck"
```
Comment (zpao) [inline .circleci/config.yml:1232]: Why do we need to set the working directory? Aren't we just working from the root of the clone?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1277,3 +1307,9 @@ workflows:
               only: /^v20.*/
             branches:
               ignore: /.*/
+      - publish_docs:
+          filters:
+            tags:
+              only: /^v20.*/
+            branches:
+              ignore: /.*/
```
Comment (zpao) [inline .circleci/config.yml:1315]: Lot of copy & paste of these filters… might make sense to share them and `<<` include them.

File: docs/publish.sh [hunk 3]
```diff
@@ -101,6 +103,8 @@ cd -
 
 # Commit the new version of the docs:
 cd $STATIC_FILES_DIR
+echo "${CNAME}" > CNAME
```
Comment (zpao) [inline docs/publish.sh:106]: In an ideal world this wouldn't be something that happens magically in bash, but is actually a file that lives in the website folder. This is fine for now though

File: .circleci/config.yml (hunk 1 above)
Comment (v-jizhang) [reply .circleci/config.yml:1232]: This instructs CircleCI where to clone the code to.

**Pull request merged**